packages/hydrogen-react/src/ImageLegacy.tsx

```diff
@@ -0,0 +1,220 @@
+import * as React from 'react';
+import {
+  getShopifyImageDimensions,
+  shopifyImageLoader,
+  addImageSizeParametersToUrl,
+  IMG_SRC_SET_SIZES,
+} from './image-size.js';
+import type {Image as ImageType} from './storefront-api-types.js';
+import type {PartialDeep, Simplify} from 'type-fest';
+
+type HtmlImageProps = React.ImgHTMLAttributes<HTMLImageElement>;
+
+export type ShopifyLoaderOptions = {
+  crop?: 'top' | 'bottom' | 'left' | 'right' | 'center';
+  scale?: 2 | 3;
+  width?: HtmlImageProps['width'] | ImageType['width'];
+  height?: HtmlImageProps['height'] | ImageType['height'];
+};
+export type ShopifyLoaderParams = Simplify<ShopifyLoaderOptions & ImageSrc>;
+
+type ImageSrc = {
+  src: ImageType['url'];
+};
+
+export type ShopifyImageProps = Omit<HtmlImageProps, 'src'> &
+  ShopifyImageBaseProps;
+
+type ShopifyImageBaseProps = {
+  /** An object with fields that correspond to the Storefront API's
+   * [Image object](https://shopify.dev/api/storefront/reference/common-objects/image).
+   * The `data` prop is required.
+   */
+  data: PartialDeep<ImageType, {recurseIntoArrays: true}>;
+  /** A custom function that generates the image URL. Parameters passed in
+   * are `ShopifyLoaderParams`
+   */
+  loader?: (params: ShopifyLoaderParams) => string;
+  /** An object of `loader` function options. For example, if the `loader` function
+   * requires a `scale` option, then the value can be a property of the
+   * `loaderOptions` object (for example, `{scale: 2}`). The object shape is `ShopifyLoaderOptions`.
+   */
+  loaderOptions?: ShopifyLoaderOptions;
+  /**
+   * `src` isn't used, and should instead be passed as part of the `data` object
+   */
+  src?: never;
+  /**
+   * An array of pixel widths to overwrite the default generated srcset. For example, `[300, 600, 800]`.
+   */
+  widths?: (HtmlImageProps['width'] | ImageType['width'])[];
+};
+
+/**
+ * The `Image` component renders an image for the Storefront API's
```

Comment:
If you add the `@deprecated` comment here it should help users with warnings in their IDEs.